library/src/main/java/com/bumptech/glide/load/engine/Engine.java

```diff
@@ -293,6 +293,10 @@ public void onResourceReleased(Key cacheKey, EngineResource resource) {
         }
     }
 
+    public void clearDiskCache() {
+        diskCacheProvider.getDiskCache().clear();
```

Comment:
I think I'm not 100% clear on the use case. If you're trying to make sure all entries are removed, there's no guarantee (asynchronous or not) that in flight requests won't still complete having loaded data from the cache after this method is called. 

Can anyone share a use case or two? What's the goal of clearing the cache? 

I'm still inclined to say a version number might be the best way to do this, because it would at least let you provide some guarantees (ie no request will complete with old data if you pass in an incremented disk cache version number to Glide as part of the GlideBuilder). Again though I'm not 100% clear on the use case.